packages/build/src/plugins_core/deploy_config/index.ts

```diff
@@ -0,0 +1,40 @@
+import { promises as fs } from 'fs'
+import { resolve } from 'path'
+
+import merge from 'deepmerge'
+
+import type { NetlifyConfig } from '../../index.js'
+import { CoreStep, CoreStepFunction } from '../types.js'
+
+const coreStep: CoreStepFunction = async function ({ buildDir, netlifyConfig }) {
```

Comment:
Not a blocker but thinking it could be useful to leverage our traces here and add some data on the fields that changed (and potentially the values if these don't hold sensitive info). This could help us with future headaches around debugging config changes or spot potential inconsistencies. We have other core steps already creating some spans we can use as boilerplate:
- https://github.com/netlify/build/blob/ba81c85aad902380d24fc1722bc65d4feed24b84/packages/build/src/plugins_core/secrets_scanning/index.ts#L69-L91

Alternatively we could just add some events on the current span (we already create a [span per execution step](https://github.com/netlify/build/blob/ba81c85aad902380d24fc1722bc65d4feed24b84/packages/build/src/steps/run_step.ts#L85)):
- https://opentelemetry.io/docs/languages/js/instrumentation/#span-events
- https://opentelemetry.io/docs/languages/js/instrumentation/#get-a-span-from-context

Reply:
I haven't added traces yet, but I did add a system logger. Will definitely revisit, though.